src/traits.rs

```diff
@@ -0,0 +1,17 @@
+pub trait Something {
+    fn get_name(&self) -> &str;
+}
+
+pub trait Entity : Something {
+    fn get_function(&self) -> &str;
+
+    fn summon(&self) {
+        println!("I summon you {0} ! All hail {0}, {1} !", self.get_name(), self.get_function());
+    }
+}
+
+pub trait Expendable : Something {
+    fn sacrifice_to<T: Entity>(&self, entity: &T) {
```

Comment:
Clearly, `self` should be taken by vale here. I do not expect to be able to sacrifice the same thing two times.